watertap/tools/parameter_sweep.py

```diff
@@ -771,6 +771,63 @@ def _read_output_h5(filepath):
 # ================================================================
 
 
+def _param_sweep_kernel(
+    model,
+    optimize_function,
+    optimize_kwargs,
+    reinitialize_before_sweep,
+    reinitialize_function,
+    reinitialize_kwargs,
+    reinitialize_values,
+):
+
+    run_successful = False  # until proven otherwise
+
+    # Forced reinitialization of the flowsheet if enabled
+    if reinitialize_before_sweep:
+        if reinitialize_function is None:
+            raise ValueError(
+                "Reinitialization function was not specified. The model will not be reinitialized."
+            )
+        else:
+            for v, val in reinitialize_values.items():
+                if not v.fixed:
+                    v.set_value(val, skip_validation=True)
+            reinitialize_function(model, **reinitialize_kwargs)
+
+    try:
+        # Simulate/optimize with this set of parameter
+        with capture_output():
+            results = optimize_function(model, **optimize_kwargs)
+        pyo.assert_optimal_termination(results)
+
+    except:
```

Comment:
I'd recommend to specify the exception type(s) (the more specific the better).

Reply:
I don't think we should make this change, because we basically *do* want to catch everything here.

Further, this is not new code and is just reorganizing existing code into a separate function. If we do want to make this change, it should be done in a separate PR.

Reply:
```suggestion
    except RuntimeError:
```

Reply:
@bknueven: :man_shrugging: :+1:

Reply:
This is not things raised by `assert_optimal_termination`, per se, because otherwise we would use `check_optimal_termination`.

Sometimes the parameter sweep tool finds weird ways to make either Ipopt *or* Pyomo, *or* the combination of both fail in ways that are difficult to both find and enumerate (e.g., inside the `optimize_function`). So any change to this blanket `except` needs to be very carefully considered.

Reply:
@bknueven I see, thanks for the clarification and the extra context. I agree that the "catch-all" approach is more justified here.